### ISSUE TITLE
When non-zero 'scrolloff' is used, starting caret position of first window is offscreen

### ISSUE THREAD

**Issue by ricksladkey** [jaredpar/VsVim#2240]
I have been trying to as an experiment to use `:set scrolloff=2` as my normal VsVim editor setting and trying to observe any misbehavior. One odd symptom that I finally associated with this setting is that upon starting Visual Studio, the caret of the restored active document was often offscreen. This only seems to affect the very first document. Other lazy loaded documents appear correctly, that is "how they last appeared the last time that Visual Studio was previously closed."

In other words, the active document's buffer window was properly scrolled when Visual Studio was closed (more than `scrolloff` away from the window's top and bottom), but upon restarting Visual Studio, the active document scroll position is "warped" from its remembered position to quite the opposite of what `scrolloff` is intended to do, scrolled so that the caret position is just offscreen.
